pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DownloadParticipantListRoute.java

```diff
@@ -0,0 +1,181 @@
+import spark.Request;
+import spark.Response;
+
+public class DownloadParticipantListRoute extends RequestHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(DownloadParticipantListRoute.class);
+    private static final String FILE_DATE_FORMAT = "ddMMyyyy";
+
+    public DownloadParticipantListRoute() {
+    }
+
+    @Override
+    public Object processRequest(Request request, Response response, String userId) throws Exception {
+        DownloadParticipantListPayload payload = new Gson().fromJson(request.body(), DownloadParticipantListPayload.class);
+
+        List<ParticipantColumn> columnNames = payload.getColumnNames();
+        List<String> esAliases = columnNames.stream()
+                .map(column -> {
+                    Alias esAlias;
+                    if (column.getObject() != null) {
+                        esAlias = Alias.of(column.getObject());
+                    } else {
+                        esAlias = Alias.of(column.getTableAlias());
+                    }
+                    return esAlias.getValue().isEmpty()? column.getName() : esAlias.getValue() + "." + column.getName();
+                })
+                .collect(Collectors.toList());
+
+        Filterable filterable = FilterFactory.of(request);
+        ParticipantWrapperResult filteredList = (ParticipantWrapperResult) filterable.filter(request.queryMap());
+        Workbook workbook = new XSSFWorkbook();
+        List<String> headerNames = payload.getHeaderNames();
+        Sheet sheet = workbook.createSheet("Participant List");
+        Row header = sheet.createRow(0);
+        IntStream.range(0, headerNames.size()).forEach(i -> {
+            Cell cell = header.createCell(i);
+            cell.setCellValue(headerNames.get(i));
+            sheet.autoSizeColumn(i);
+        });
+        int currentRow = 1;
+        for (ParticipantWrapperDto participant : filteredList.getParticipants()) {
+            List<Object> excelRow = new ArrayList<>();
+            Map<String, Object> esDataAsMap = participant.getEsDataAsMap();
+            List<String> row = new ArrayList<>();
+            for (String esAlias : esAliases) {
+                Object nestedValue = getNestedValue(esAlias, esDataAsMap);
+                excelRow.add(nestedValue);
+                if (nestedValue instanceof Collection) {
+                    Collection<?> value = (Collection<?>) nestedValue;
+                    row.add(value.stream().map(Object::toString)
+                            .collect(Collectors.joining(System.lineSeparator())));
+                } else {
+                    row.add(nestedValue.toString());
+                }
+            }
+            currentRow = createRecord(sheet, currentRow, new ParticipantRecord(excelRow));
+        }
+        responseFile(response, workbook);
+        return response.raw();
+    }
+
+
+    private void responseFile(Response response, Workbook workbook) throws Exception {
+        File currDir = new File(".");
```

Comment:
having here `try-with-resource` statement to handle `AutoCloseable` type objects (`Workbook`, `FileOutputStream`, `ServletOutputStream`) instead of closing them manually would be better.